### PR TITLE
Learning path automation

### LINKED ISSUE #349: Automate publishing to innersourcecommons.org
Every time there is a new article or a new contributor we need to manually run [these steps](https://github.com/InnerSourceCommons/InnerSourceLearningPath/tree/master/scripts#how-to-update-innersourcecommonsorg-with-new-articles) to publish on innersourcecommons.org.

We should automate this process so that it happens automatically every time there is a change to an article in the learning path.  Likely this should be done with GitHub actions.

### PR DESCRIPTION
**Pull request by tsadler1988**
Along with https://github.com/InnerSourceCommons/innersourcecommons.org/pull/130, this fixes #349.

This new GitHub Action will (if relevant files have changed on master):
- Checkout LearningPath and innersourcecommons.org repos
- Checkout or create a `publish-latest-learning-path` branch of innersourcecommons.org
- Run the scripts to generate website markdown from learning path articles and data
- Commit and push to the `publish-latest-learning-path` branch of innersourcecommons.org

Opening a Pull Request will be done a separate action run on innersourcecommons.org repo, due to limitations of GitHub deploy keys (see https://github.com/InnerSourceCommons/innersourcecommons.org/pull/130).

This has been tested against earlier commits, where the action was configured to build from this branch - https://github.com/InnerSourceCommons/InnerSourceLearningPath/actions/workflows/publish-to-website.yml - and the output used to open a PR updating the website - https://github.com/InnerSourceCommons/innersourcecommons.org/pull/134.

### CODE REVIEW

**Comment by tsadler1988** [comment]
@lenucksi / @rrrutledge would either or both of you be able to review this please?

File: config/urls.yaml [hunk 1]
```diff
@@ -226,7 +226,7 @@
   video:
     isc: https://innersourcecommons.org/learn/learning-path/contributor/02/
     oreilly: https://learning.oreilly.com/learning-paths/learning-path-the/0636920338833/0636920338802-video329500
-    youtube: https://youtu.be/S0Gps2AbZ7M
+    youtube: https://www.youtube.com/watch?v=S0Gps2AbZ7M
```
Comment (rrrutledge) [inline config/urls.yaml:229]: Just curious why this change?

File: .github/workflows/publish-to-website.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Learning Path
+        uses: actions/checkout@v2
+        with:
+          path: ./InnerSourceLearningPath
+      - name: Checkout innersourcecommons.org
+        uses: actions/checkout@v2
+        with:
+          ssh-key: ${{ secrets.WEBSITE_DEPLOY_KEY }}
```
Comment (rrrutledge) [inline .github/workflows/publish-to-website.yml:31]: This is entered in by repo admins somewhere in the UI?

File: .github/workflows/publish-to-website.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Build
+        working-directory: InnerSourceLearningPath/scripts
+        run: |
+            npm ci
+            TOKEN=${{ secrets.GITHUB_TOKEN }} node generate_learning_path_markdown.js
+            cp -r newsite/. ../../innersourcecommons.org/content/learn/learning-path/
+      - name : Commit
```
Comment (rrrutledge) [inline .github/workflows/publish-to-website.yml:49]: Is there a way to open a PR or do we still have to do that manually?

Comment (rrrutledge) [reply .github/workflows/publish-to-website.yml:49]: Maybe we should just list directly to _main_ and skip the whole branch thing.

**Review by rrrutledge — APPROVED** [review]
Very neat to see!

File: config/urls.yaml (hunk 1 above)
Comment (tsadler1988) [reply config/urls.yaml:229]: When this was updated in #431 it used the `youtu.be` format, but the script expects `www.youtube.com` format - all other YouTube URLS in this file are in that format. The script needs to extract the video ID from the URL.

File: .github/workflows/publish-to-website.yml (hunk 2 above)
Comment (tsadler1988) [reply .github/workflows/publish-to-website.yml:31]: Correct. @voborgus added the public key as a Deploy Key  for https://github.com/InnerSourceCommons/innersourcecommons.org, and I added the private key to https://github.com/InnerSourceCommons/InnerSourceLearningPath as an Action secret.

File: .github/workflows/publish-to-website.yml (hunk 3 above)
Comment (tsadler1988) [reply .github/workflows/publish-to-website.yml:49]: Discussed in https://github.com/InnerSourceCommons/innersourcecommons.org/pull/130, but to summarise:
- Deploy Keys do not have GitHub API access, so this Action cannot raise a PR. Instead, there is another Action on the innersourcecommons.org repo which raises a PR whenever the `publish-latest-learning-path` branch is pushed to (added in https://github.com/InnerSourceCommons/innersourcecommons.org/pull/130). We considered alternatives to using a Deploy Key, but stuck with this solution.
- Original plan was to push directly to main, but this would trigger a website rebuild into production and the preference was to have a manual step to reduce the risk of an outage of the website.
